website/src/content/blog/2025-03-31-typespec-1-0-release/type_spec_1_0_release.md

```diff
@@ -0,0 +1,188 @@
+- **<a href="https://www.npmjs.com/package/@typespec/http-client-js" target="_blank" rel="noopener noreferrer">@typespec/http-client-js</a>**
+- **<a href="https://www.npmjs.com/package/@typespec/http-client-java" target="_blank" rel="noopener noreferrer">@typespec/http-client-java</a>**
+- **<a href="https://www.npmjs.com/package/@typespec/http-client-python" target="_blank" rel="noopener noreferrer">@typespec/http-client-python</a>**
+- **<a href="https://www.npmjs.com/package/@typespec/http-server-csharp" target="_blank" rel="noopener noreferrer">@typespec/http-server-csharp</a>**
+- **<a href="https://www.npmjs.com/package/@typespec/http-server-js" target="_blank" rel="noopener noreferrer">@typespec/http-server-js</a>**
+
+## Getting started with TypeSpec 1.0-RC
+
+If you just want to jump in, follow our <a href="https://typespec.io/docs/" target="_blank" rel="noopener noreferrer">install guide</a>, define your models and operations, and generate API specifications with our tools. Visit <a href="https://typespec.io/docs/" target="_blank" rel="noopener noreferrer">our docs page</a> for guides and examples, or join our <a href="https://typespec.io/community/" target="_blank" rel="noopener noreferrer">community</a> to connect with other TypeSpec users.
+
+If you want to know more, keep reading!
+
+## Emitters: From TypeSpec to the API format you need
+
+TypeSpec's core value proposition is the ability to define your API once and generate multiple output formats. The 1.0-RC release ships with an OpenAPI 3.0 emitter, <a href="https://www.npmjs.com/package/@typespec/openapi3" target="_blank" rel="noopener noreferrer">@typespec/openapi3</a>, enabling you to seamlessly integrate TypeSpec into existing OpenAPI workflows and toolchains.
+
+For teams with existing OpenAPI definitions looking to adopt TypeSpec, we've also created an <a href="https://typespec.io/docs/emitters/openapi3/cli/" target="_blank" rel="noopener noreferrer">OpenAPI to TypeSpec converter tool</a>. This tool helps you migrate your existing API definitions to TypeSpec, providing a smoother onboarding experience and preserving your investment in OpenAPI.
+
+## Code generation: From TypeSpec to server & client code
+
+TypeSpec 1.0-RC introduces a powerful code generation feature (in preview) that transforms how you build APIs. Define your API once in TypeSpec, and automatically generate both client libraries and server implementations across multiple programming languages. This eliminates repetitive boilerplate code, accelerates development, and lets you focus exclusively on your core business logic.
+
+![TypeSpec Workflow Diagram](./workflow-diagram-full.png)
+
+### Server-side code generation
+
+Imagine building a "to do" service API. With TypeSpec, you start by defining your data models and operations in a concise and human-readable format. For example:
+
+```tsp
+@route("/todoitems")
+namespace TodoItems {
```

Comment:
With the models removed I'd use interface here.

Reply:
How would you write this using an interface?